### PR TITLE
chore: remove some clones

### PR DESCRIPTION
**Pull request by hai-rise**

### CODE REVIEW

**Review by Rjected — APPROVED** [review]
lgtm

File: crates/optimism/payload/src/builder.rs [hunk 1]
```diff
@@ -156,7 +150,7 @@ where
 /// a result indicating success with the payload or an error in case of failure.
 #[inline]
 pub(crate) fn optimism_payload<EvmConfig, Pool, Client>(
-    evm_config: EvmConfig,
+    evm_config: &EvmConfig,
```
Comment (mattsse) [inline crates/optimism/payload/src/builder.rs:153]: this is not strictly required
```suggestion
    evm_config: EvmConfig,
```

**Review by mattsse — APPROVED** [review]
lgtm,
one pedantic nit

File: crates/optimism/payload/src/builder.rs (hunk 1 above)
Comment (mattsse) [reply crates/optimism/payload/src/builder.rs:153]: EvmConfig is auto_impl for & so this shouldn't be required by we can also do &

Comment (hai-rise) [reply crates/optimism/payload/src/builder.rs:153]: Wouldn't work :sweat_smile:.

```
error[E0521]: borrowed data escapes outside of method
   --> crates/optimism/payload/src/builder.rs:108:9
    |
104 |         &self,
    |         -----
    |         |
    |         `self` is a reference that is only valid in the method body
    |         let's call the lifetime of this reference `'1`
...
108 |         optimism_payload(&self.evm_config, args, cfg_env, block_env, self.compute_pending_block)
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |         |
    |         `self` escapes the method body here
    |         argument requires that `'1` must outlive `'static`
```

**Pull request merged**